src/V3Split.cpp

```diff
@@ -791,21 +791,54 @@ class EmitSplitVisitor final : public VNVisitor {
 };
 
 class RemovePlaceholdersVisitor final : public VNVisitor {
-    std::unordered_set<AstNode*> m_removeSet;  // placeholders to be removed
-public:
-    explicit RemovePlaceholdersVisitor(AstNode* nodep) {
-        iterate(nodep);
-        for (AstNode* np : m_removeSet) {
-            np->unlinkFrBack();  // Without next
-            VL_DO_DANGLING(np->deleteTree(), np);
+    // MEMBERS
+    bool m_isPure = true;
+    int m_emptyAlways = 0;
+
+    // CONSTRUCTORS
+    RemovePlaceholdersVisitor() = default;
+    virtual ~RemovePlaceholdersVisitor() override = default;
+
+    // VISITORS
+    virtual void visit(AstSplitPlaceholder* nodep) override { pushDeletep(nodep->unlinkFrBack()); }
+    virtual void visit(AstNodeIf* nodep) override {
+        VL_RESTORER(m_isPure);
+        m_isPure = true;
+        iterateChildren(nodep);
+        if (!nodep->ifsp() && !nodep->elsesp() && m_isPure) pushDeletep(nodep->unlinkFrBack());
+    }
+    virtual void visit(AstAlways* nodep) override {
+        m_isPure = true;
```

Comment:
```suggestion
        VL_RESTORER(m_isPure);
        m_isPure = true;
```

(not that it really matters, but is cleaner)